### DISCUSSION TITLE
kernel die while importing MPRester/matproj

### DISCUSSION THREAD

**Discussion by sharma1420** [materialsproject/pymatgen#3391] [Q&A] [answered]
May you please help me by checking error with MPRester,

I have just updated system to MacBook Pro M2. previously I was using intel - MacBook Air, I was able to use perfectly Pymatgen and MPRester.
But in the new MacBook M2, I am not able to import MPRester, kernel get dies, each time I am importing MPRester. **from pymatgen.ext.matproj import MPRester** ( I checked its working fine till .ext but as soon as I import **matproj** its not working)

**Comment by janosh** [comment]
We need the script you're running and installed python + package versions.

**Comment by sharma1420** [comment]
Dear janosh, 
I attached image of script I am running. I have Python3.11 and new version of  Pymatgen (Oct '23)
<img width="906" alt="Screenshot 2023-10-10 at 9 45 12 AM" src="https://github.com/materialsproject/pymatgen/assets/118822279/7de3f621-ee25-4a3d-b1cb-05feea4eefc5">

**Comment by sharma1420** [comment, marked as answer]
It resolved @janosh , [as per the suggestion](https://matsci.org/t/kernel-die-while-importing-mprester-on-m2-pro/51197) by @tschaume I updated  Python to 3.10, and it's working fine now.

**Reply by chhetrimanoj1992** [reply]
Hi there, I am also having exact same issue in my Mac M1. So the solution is to change python to 3.10? I have python 3.11.8. Please reply.

**Reply by sharma1420** [reply]
yes @manojchh58 ,it worked for me I have M2 version.

**Reply by chhetrimanoj1992** [reply]
So I need to downgrade python from 3.11.8 to 3.10?

**Reply by sharma1420** [reply]
Please check and, update if its working ?

**Reply by chhetrimanoj1992** [reply]
Here is my details:

I am facing a problem that whenever I tried to import MPRester from either pymatgen (from pymatgen.ext.matproj import MPRester) or (from mp_api.client import MPRester), an error appears saying kernel died.

- I used virtual env (activating) to perform this. 
- I installed pymatgen, python3, matminer and other necessary packeages in virtual env
- then I enter to jupyter notebook from virtual env
- pymatgen and matminer got imported without error but "kernel died" when I start importing MPRester. 

Specifications:
- device Mac M1, SONOMA 14.2.1
- python 3.11.8
- pymatgen 2024.3.1
- matminer 0.9.0
- conda 24.1.2
- jupyter notebook 7.1.2

@tschaume @sharma1420

**Reply by chhetrimanoj1992** [reply]
Hi, yes it worked. I updated python to 3.12.2. Thank you so much. Also, I want to stay in touch and updated on what you currently doing in this field. Please send me an email: chhetrimanoj1992@gmail.com
